src/pint/models/stand_alone_psr_binaries/binary_generic.py

```diff
@@ -421,7 +417,7 @@ def d_a1_d_par(self, par):
         par_obj = getattr(self, par)
         try:
             func = getattr(self, f"d_a1_d_{par}")
-        except:
+        except Exception:
```

Comment:
Would `AttributeError` be better here (if it can't find `self.d_a1_d_par`?)

Reply:
Yes. Changed.